src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataPrimitiveDeserializer.cs

```diff
@@ -66,8 +66,13 @@ public virtual object ReadPrimitive(ODataProperty primitiveProperty, ODataDeseri
             {
                 throw Error.ArgumentNull("primitiveProperty");
             }
+            //Try and change the value appropriately if type is specified
+            if (readContext != null && readContext.ResourceType != null && primitiveProperty.Value != null)
```

Comment:
>readContext != null [](start = 16, length = 19)

Do we allow "readContext== null" ?

if not, maybe it's better to add the argument check, like the "PrimitiveProperty"

Reply:
Added the readContext null check.